src/main/java/podo/odeego/infra/openapi/naver/NaverClient.java

```diff
@@ -0,0 +1,98 @@
+package podo.odeego.infra.openapi.naver;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import podo.odeego.domain.place.domain.PlaceCategory;
+import podo.odeego.domain.place.dto.SimplePlaceResponse;
+import podo.odeego.infra.openapi.naver.dto.LocalSearchRequest;
+import podo.odeego.infra.openapi.naver.dto.LocalSearchResponse;
+
+@Component
+public class NaverClient {
+
+	@Value("${naver.header.id}")
+	private String headerId;
+
+	@Value("${naver.header.secret}")
+	private String headerSecret;
+
+	@Value("${naver.client.id}")
+	private String clientId;
+
+	@Value("${naver.client.secret}")
+	private String clientSecret;
+
+	@Value("${naver.url.search.local}")
+	private String localSearchUrl;
+
+	public List<SimplePlaceResponse> searchLocal(String query) {
+		return Arrays.stream(PlaceCategory.values())
+			.map(category -> searchLocal(query, category))
+			.flatMap(List::stream)
+			.distinct()
+			.toList();
+	}
+
+	public List<SimplePlaceResponse> searchLocal(String query, PlaceCategory category) {
+		return getLocalSearchRequests(query, category).stream()
+			.map(this::callLocalSearchApi)
+			.map(LocalSearchResponse::getPlaces)
+			.flatMap(List::stream)
+			.distinct()
+			.toList();
+	}
+
+	private List<LocalSearchRequest> getLocalSearchRequests(String query, PlaceCategory category) {
```

Comment:
A: 사실상 LocalSearchRequest를 생성하는 역할을 하고 있네요. 객체를 생성하는 뜻을 주면 어떨까요?
```suggestion
	private List<LocalSearchRequest> generateLocalSearchRequests(String query, PlaceCategory category) {
```

Reply:
좋습니다. 확실히 기존에 있는 객체를 가져오냐, 객체를 새로 생성하냐를 분명하게 구분하는 것이 사용하는 입장에서 더 명확할 것 같습니다. :)